src/coreclr/src/vm/comthreadpool.cpp

```diff
@@ -446,6 +555,32 @@ FCIMPL5(LPVOID, ThreadPoolNative::CorRegisterWaitForSingleObject,
 }
 FCIMPLEND
 
+#ifdef TARGET_WINDOWS // the IO completion thread pool is currently only available on Windows
+FCIMPL1(void, ThreadPoolNative::CorQueueWaitCompletion, Object* completeWaitWorkItemObjectUNSAFE)
+{
+    FCALL_CONTRACT;
+    _ASSERTE_ALL_BUILDS(__FILE__, ThreadpoolMgr::UsePortableThreadPool());
+
+    HANDLE completeWaitWorkItemHandle = NULL;
+    struct _gc
```

Comment:
A nit - since you need to protect just a single object, you can use the HELPER_METHOD_FRAME_BEGIN_1 version without creating the _gc struct. There are also variants for 2 and 3 objects, only when using more objects you need to use the HELPER_METHOD_FRAME_BEGIN_PROTECT.
But please feel free to keep it as is if you prefer it that way.